### PR TITLE
fix(bruteforce-protection): Don't throw a 500 when MaxDelayReached is…

### PR DESCRIPTION
**Pull request by nickvergessen**
… thrown

## Test instructions

```diff
diff --git a/index.php b/index.php
index f9756b7b51a..bbadeb4970c 100644
--- a/index.php
+++ b/index.php
@@ -36,6 +36,7 @@ use Psr\Log\LoggerInterface;
 try {
 	require_once __DIR__ . '/lib/base.php';
 
+	throw new MaxDelayReached('Reached maximum delay');
 	OC::handleRequest();
 } catch (\OC\ServiceUnavailableException $ex) {
 	\OC::$server->get(LoggerInterface::class)->error($ex->getMessage(), [
diff --git a/ocs/v1.php b/ocs/v1.php
index 55e9f426aba..cc115d0e352 100644
--- a/ocs/v1.php
+++ b/ocs/v1.php
@@ -58,6 +58,7 @@ try {
 	// side effects in existing apps
 	OC_App::loadApps();
 
+	throw new MaxDelayReached('Reached maximum delay');
 	if (!\OC::$server->getUserSession()->isLoggedIn()) {
 		OC::handleLogin(\OC::$server->getRequest());
 	}
```

### Tested with:

- Frontpage: `curl -k https://nextcloud29.local/index.php/cloud/user?format=json -u admin:admin -i -H 'OCS-APIRequest: true' -H 'Accept: text/html'`
- index.php API call: `curl -k https://nextcloud29.local/index.php/cloud/user?format=json -u admin:admin -i -H 'OCS-APIRequest: true'`
- OCS API call: `curl -k https://nextcloud29.local/ocs/v2.php/cloud/user?format=json -u admin:admin -i -H 'OCS-APIRequest: true'`

### Before

```
$ curl -k https://nextcloud29.local/ocs/v2.php/cloud/user?format=json -u admin:admin -i -H 'OCS-APIRequest: true'
HTTP/1.1 500 Internal Server Error
…
{"ocs":{"meta":{"status":"failure","statuscode":500,"message":"Internal Server Error\nReached maximum delay"},"data":[]}}
```

### After
```
$ curl -k https://nextcloud29.local/ocs/v2.php/cloud/user?format=json -u admin:admin -i -H 'OCS-APIRequest: true'
HTTP/1.1 429 Too Many Requests
…
{"ocs":{"meta":{"status":"failure","statuscode":429,"message":"Reached maximum delay"},"data":[]}}
```

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -77,6 +79,21 @@
 		exit();
 	}
 	OC_Template::printErrorPage($ex->getMessage(), $ex->getMessage(), 401);
+} catch (MaxDelayReached $ex) {
+	$request = \OC::$server->getRequest();
+	/**
+	 * Routes with the @CORS annotation and other API endpoints should
+	 * not return a webpage, so we only print the error page when html is accepted,
+	 * otherwise we reply with a JSON array like the SecurityMiddleware would do.
+	 */
```
Comment (blizzz) [inline index.php:88]: The comment looks out of place

**Review by miaulalala — APPROVED** [review]

File: index.php (hunk 1 above)
Comment (nickvergessen) [reply index.php:88]: It's copy-pasted from the LoginException which should behave very similar, that's why I copied it.

Comment (blizzz) [reply index.php:88]: I thought you copied it, but the content does not make much sense in this context, does it?

**Review by AndyScherzinger — APPROVED** [review]

**Review by blizzz — APPROVED** [review]

**Pull request merged**